### DISCUSSION TITLE
AWS Lamda Handler Type

### DISCUSSION THREAD

**Discussion by makmashuk** [aws/aws-sdk-js-v3#5611] [General]
Hey, we are migrating aws sdk to v3 but for lamda handler we couldn't find any type inside @aws-sdk/client-lambda package, Are you planning to add the handler types inside the package or we need to use @types/aws-lambda?

**Comment by aungkhantkyaw9** [comment]
I am also have the same issue. In v2 there are types for aws-lambda. But in v3, could not find any type related to AWS Lambda functions

**Comment by dmoskovtsov** [comment]
Example for v2: 



Can anyone please provide information on the expected way of typing it in v3? (using "@aws-sdk/client-lambda": "3.476.0")

```import { SQSHandler, SNSHandler, APIGatewayProxyHandler } from 'aws-lambda';

export const apiV2Handler: APIGatewayProxyHandler = async (event, context) => {
    return {
        body: 'Response body',
        statusCode: 200
    }
}

**Comment by RanVaknin** [comment]
Hi everyone on the thread. 

Can you highlight specific imports that are missing from v3 but existed in v2? A complete code sample with imports would be ideal.

Thanks,
Ran~

**Comment by makmashuk** [comment]
Hey , Please check the below block 
```js
import { Handler } from 'aws-lambda';

export const handler: Handler = async (event, context) => {
    console.log('EVENT: \n' + JSON.stringify(event, null, 2));
    return context.logStreamName;
};

import { Context, APIGatewayProxyCallback, APIGatewayEvent } from 'aws-lambda';

export const lambdaHandler = (event: APIGatewayEvent, context: Context, callback: APIGatewayProxyCallback): void => {
    console.log(`Event: ${JSON.stringify(event, null, 2)}`);
    console.log(`Context: ${JSON.stringify(context, null, 2)}`);
    callback(null, {
        statusCode: 200,
        body: JSON.stringify({
            message: 'hello world',
        }),
    });
};
```
currently we were using this package(https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/aws-lambda) for AWS SDK v2

**Reply by RanVaknin** [reply]
Thanks for the context. 

I just want to highlight that `aws-lambda` is an independent 3rd party package that is not maintained by AWS. As far as I can tell, it's not scoped to the v2 SDK, and does not seem to be related to any SDK for that matter, this is just a library of types.

Neither v2, nor v3 offer type definitions for writing Lambda functions. The type definitions the official AWS SDK `@aws-sdk/client-lambda` offers is for the Lambda's control plane API, which means the types generated are for provisioning, updating, deleting and invoking functions, and _not for writing the lambda code itself_. 

Unlike https://github.com/aws/aws-lambda-go, which is an official package managed by the Lambda team on AWS, there is no JS equivalent I am aware of, therefore I cannot make recommendations about this. 

If this has worked for you in v2, I don't see a reason why this would not work on v3, as this is independent from the AWS SDK offering.

Thanks,
Ran~

**Reply by couimet** [reply]
Hello @RanVaknin !

Now that close to 6 months has passed, I wanted to validate that the above answer is still valid.

AWS is still **not** providing types definitions for _writing lambda code itself_ ?

Given that https://aws.amazon.com/blogs/developer/first-class-typescript-support-in-modular-aws-sdk-for-javascript/ is titled `First-class TypeScript support in modular AWS SDK for JavaScript`, I was somewhat expecting that AWS would provide an equivalent of https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/aws-lambda so we do not need to rely on a 3rd party as the source of the TypeScript types ?
